lib/users.js

```diff
@@ -14,23 +17,164 @@ const db = require('./db')
  * @returns {user object} User object (containing name)
  */
 function get (token) {
-  const sql = 'select * from user_tokens where token=$1'
+  const sql = 'SELECT * FROM user_tokens WHERE token=$1'
   return db.oneOrNone(sql, [token])
 }
 
 /**
- * Get multiple users given an array of tokens
+ * Get multiple users given an array of ids
  *
  * @name getByIds
  * @function
  *
- * @param {array} tokens Array with users' tokens
+ * @param {array} ids Array with users' ids
  *
  * @returns {array} Array of users found
  */
-function getByIds (tokens) {
-  const sql = 'select * from user_tokens where token in ($1^)'
-  const tokensClause = _.map(pgp.as.text, tokens).join(',')
-  return db.any(sql, [tokensClause])
+function getByIds (ids) {
+  const sql = `SELECT * FROM users WHERE id IN ($1^)`
+  const idList = _.map(pgp.as.text, ids).join(',')
+  return db.any(sql, [idList])
+}
+
+function getUserById (id) {
+  const sql = `SELECT * FROM users WHERE id=$1`
+  return db.oneOrNone(sql, [id])
+}
+
+function getUserByUsername (username) {
+  const sql = `SELECT * FROM users WHERE username=$1`
+  return db.oneOrNone(sql, [username])
+}
+
+function getUsers () {
+  const sql = `SELECT id, username, role, enabled, last_accessed, last_accessed_from, last_accessed_address FROM users ORDER BY username`
+  return db.any(sql)
+}
+
+function verifyAndUpdateUser (id, ua, ip) {
```

Comment:
is this being used?

Reply:
Yes, it is used in the apollo context, to confirm if the cookie corresponds to an existing user and update the ip address, user agent and such of both the DB and the cookie